src/bigistore.c

```diff
@@ -1099,6 +1099,58 @@ Datum bigistore_slice_to_array(PG_FUNCTION_ARGS)
     PG_RETURN_POINTER(aout);
 }
 
+static void bigistore_clamp_pass(BigIStore * is, int32 end_key, int delta_dir)
+{
+    BigIStorePair * pairs;
+    int64 result = 0;
+    int   index  = 0, count = 0;
+
+    pairs = FIRST_PAIR(is, BigIStorePair);
+    index = delta_dir > 0 ? 0 : is->len - 1;
+    while ( ((delta_dir > 0) && (index < is->len && pairs[index].key <= end_key)) ||
```

Comment:
it's not wrong but excluding the equality case would save a loop call
and avoid setting the index back later

Reply:
If `key == end_key` is present in the `istore` we actually need both, add it to the result and decrease `buflen`.

Reply:
yep but you can do it later in the if(count) i.e. `clamp_below('0=>2',0)` would run into unnecessary code

Reply:
I'd rather leave the loop alone, and not call into `clamp_pass` at all, as you have suggested in another thread